contracts/DAOTreasury.cdc

```diff
@@ -62,9 +97,18 @@ pub contract DAOTreasuryV3 {
     pub fun executeAction(actionUUID: UInt64, signaturePayload: MyMultiSigV3.MessageSignaturePayload) {
       self.validateTreasurySigner(identifier: actionUUID.toString(), signaturePayload: signaturePayload)
 
+      let action = self.multiSignManager.borrowAction(actionUUID: actionUUID)
+      let actionView = action.getView()
       let selfRef: &Treasury = &self as &Treasury
+      emit ActionExecuted(
+        treasuryUUID: self.uuid,
+        actionUUID: actionUUID,
+        executor: signaturePayload.signingAddr,
+        actionView: actionView,
+        signerResponses: self.multiSignManager.getSignerResponsesForAction(actionUUID: actionUUID)
+      )
       self.multiSignManager.executeAction(actionUUID: actionUUID, {"treasury": selfRef})
-      emit ExecuteAction(actionUUID: actionUUID, proposer: signaturePayload.signingAddr)
+      
```

Comment:
i did it because the action gets destroyed so `action.signerResponses` threw an error after running `executeAction`... but i just realized i can save it to a variable before it gets destroyed, so i'll update :)